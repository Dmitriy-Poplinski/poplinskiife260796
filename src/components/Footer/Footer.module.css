.footer__main {
    position: sticky;
    height: 6%;
    display: flex;
    align-items: center;
    justify-content: center;
};
