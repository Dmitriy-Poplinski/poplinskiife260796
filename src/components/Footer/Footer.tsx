import React from 'react';

import classes from './Footer.module.css';

interface FooterProps {
    image: string
}

export function Footer({ image }: FooterProps) {
    let color;

    try {
        color = `url(${require(`../../assets/images/${image}.jpg`)})`; 
    } catch (e) {
        color = 'red'
    };

    return (
        <div 
            className={classes.footer__main}
            style={{
                background: color
            }}
            data-testid="footer"
        >
            Footer
        </div>
    );
}
