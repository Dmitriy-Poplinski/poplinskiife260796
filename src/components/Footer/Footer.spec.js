import { render, screen } from '@testing-library/react';

import { Footer } from './Footer';

describe('Footer:', () => {

    let data; 

    test('Footer gets incorrect string', () => {
        data = 'testing_image';

        render(<Footer image={data} />);

        expect(screen.getByTestId('footer')).toHaveStyle('background: red');
        expect(screen.getByTestId('footer')).toMatchSnapshot();
    });

    test('Footer gets correct string', () => {
        data = 'footer_cover';

        render(<Footer image={data} />);

        expect(screen.getByTestId('footer')).toHaveStyle(`background: url("footer_cover.jpg")`);
        expect(screen.getByTestId('footer')).toMatchSnapshot();
    });
});
