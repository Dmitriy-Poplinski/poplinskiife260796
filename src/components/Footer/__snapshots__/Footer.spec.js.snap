// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer: Footer gets correct string 1`] = `
<div
  class="footer__main"
  data-testid="footer"
  style="background: url(footer_cover.jpg);"
>
  Footer
</div>
`;

exports[`Footer: Footer gets incorrect string 1`] = `
<div
  class="footer__main"
  data-testid="footer"
  style="background: red;"
>
  Footer
</div>
`;
